### PR TITLE
Added delete methods in repositories

### LINKED ISSUE #924: [Enhancement] Add the option to delete an organization from the user's home page
## SubTasks

- The backend needs to add an API `DELETE /api/v1/organizations/{id}` that recursively deletes all the applications, pages, actions & datasources for a given organization ID. Once all the sub-entities have been deleted, the API will then delete the organization entity as well.
- The frontend needs to integrate with the API 
- The frontend needs to refresh it's local list of organizations so that this change can be reflected in the UI.

### PR DESCRIPTION
**Pull request by pc9795**
# Description
Added some deletion methods in entities related to an organization so that we can delete them when it is deleted.

Fixes #924 

## Type of change
- New feature (non-breaking change which adds functionality)

## How Has This Been Tested?

- Not required as only added spring-data-jpa methods.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Comment by pc9795** [comment]
I will create the end point once this PR is approved.

**Comment by mohanarpit** [comment]
@pc9795 I'm not sure why this change is needed. There are already functions `archive`, `archiveById` & `archiveAllById` in the `BaseRepository` interface. Can we not use those functions instead?

Traditionally, we default to using `archive` instead of JPA's delete operation because `archive` is a soft-delete. It only sets the deleted flag to true. This way we can recover documents incase there's been an erroneous delete operation.

**Comment by mohanarpit** [comment]
Sorry, I deleted my earlier comment because it was erroneous. Apologies. 

Any particular reason we are making multiple PRs to fix a single issue? Can we not do it in a single PR? I'm asking this because 
1. Tracking an issue fix to a single PR/commit makes it easier for the long term when using `git bisect` to determine where any errors were introduced. 
2. Our automation will end up closing the issue if this PR gets merged.

**Comment by pc9795** [comment]
NP! I will update the endpoint/other changes in this PR only. 

I personally prefer small PRs:
- So that small changes roll-out/validated and we don't have to revisit same PRs again and again
- A reviewer doesn't have to bang his head with changes across different parts of codebase

But for this one, I agree these two don't strongly hold and your points are correct 👍

File: app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ActionRepository.java [hunk 1]
```diff
@@ -21,4 +21,6 @@
     Flux<Action> findByPageId(String pageId);
 
     Flux<Action> findByOrganizationId(String organizationId);
+
+    Mono<Void> deleteAllByOrganizationId(String organizationId);
```
Comment (mohanarpit) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/repositories/ActionRepository.java:25]: More often than not, it's better to return the object/String (or very least; Boolean) when performing destructive operations. This ensures that the function is testable and the user also gets a visual cue on successful completion of this operation.

**Comment by mohanarpit** [comment]
Closing this PR because the change of removing a user from an organization requires a deeper change by removing their access from individual applications, actions & datasources.

**Pull request closed without merging**